pkg/gce-pd-csi-driver/controller.go

```diff
@@ -716,6 +750,95 @@ func (gceCS *GCEControllerServer) createSingleDisk(ctx context.Context, req *csi
 }
 
+func (gceCS *GCEControllerServer) diskSupportsIopsChange(diskType string) bool {
+	for _, disk := range gceCS.provisionableDisksConfig.SupportsIopsChange {
+		if disk == diskType {
+			return true
+		}
+	}
+	return false
+}
+
+func (gceCS *GCEControllerServer) diskSupportsThroughputChange(diskType string) bool {
+	for _, disk := range gceCS.provisionableDisksConfig.SupportsThroughputChange {
+		if disk == diskType {
+			return true
+		}
+	}
+	return false
+}
+
+func (gceCS *GCEControllerServer) ControllerModifyVolume(ctx context.Context, req *csi.ControllerModifyVolumeRequest) (*csi.ControllerModifyVolumeResponse, error) {
+	var err error
+
+	volumeID := req.GetVolumeId()
+	klog.V(4).Infof("Modifying Volume ID: %s", volumeID)
+
+	if volumeID == "" {
+		return nil, status.Error(codes.InvalidArgument, "volume ID must be provided")
+	}
+
+	diskType := metrics.DefaultDiskTypeForMetric
+	enableConfidentialCompute := metrics.DefaultEnableConfidentialCompute
+	enableStoragePools := metrics.DefaultEnableStoragePools
+
+	defer func() {
+		gceCS.Metrics.RecordOperationErrorMetrics("ModifyVolume", err, diskType, enableConfidentialCompute, enableStoragePools)
+	}()
+
+	project, volKey, err := common.VolumeIDToKey(volumeID)
+
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "volume ID is invalid: %v", err.Error())
+	}
+
+	volumeModifyParams, err := common.ExtractModifyVolumeParameters(req.GetMutableParameters())
+	if err != nil {
+		klog.Errorf("Failed to extract parameters for volume %s: %v", volumeID, err)
+		return nil, status.Errorf(codes.InvalidArgument, "Invalid parameters: %v", err)
+	}
+	klog.V(4).Infof("Modify Volume Parameters for %s: %v", volumeID, volumeModifyParams)
+
+	existingDisk, err := gceCS.CloudProvider.GetDisk(ctx, project, volKey, gce.GCEAPIVersionBeta)
+
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get volume: %w", err)
+	}
+
+	if existingDisk == nil || existingDisk.GetSelfLink() == "" {
+
+		return nil, status.Errorf(codes.Internal, "failed to get volume : %s", volumeID)
+	}
+	diskType = existingDisk.GetPDType()
+
+	// Check if the disk supports dynamic IOPS/Throughput provisioning
+	supportsIopsChange := gceCS.diskSupportsIopsChange(diskType)
+	supportsThroughputChange := gceCS.diskSupportsThroughputChange(diskType)
+	if !supportsIopsChange && !supportsThroughputChange {
+		return nil, status.Errorf(codes.InvalidArgument, "Failed to modify volume: modifications not supported for disk type %s", diskType)
+	}
+	if !supportsIopsChange && volumeModifyParams.IOPS != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Cannot specify IOPS for disk type %s", diskType)
+	}
+	if !supportsThroughputChange && volumeModifyParams.Throughput != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Cannot specify throughput for disk type %s", diskType)
+	}
+
+	if existingDisk.GetEnableStoragePools() {
+		enableStoragePools = "enabled"
+	} else {
+		enableStoragePools = "disabled"
+	}
+
```

Comment:
Can you replace this with 

enableStoragePools = strconv.FormatBool(existingDisk.GetEnableStoragePools())

Reply:
Done